src/pybind/mgr/cephadm/utils.py

```diff
@@ -20,6 +20,14 @@ class CephadmNoImage(Enum):
     token = 1
 
 
+def get_daemon_names(mgr: "CephadmOrchestrator", daemons: List[str]) -> List[str]:
+    daemon_names = []
+    for daemon_type in daemons:
+        for dd in mgr.cache.get_daemons_by_type(daemon_type):
+            daemon_names.append(dd.name())
+    return daemon_names
```

Comment:
now that this isn't some inline function in the calc_daemon_deps function, we might want to clean it up. Specifically I think it's a bit unintuitive that the parameter is `daemons` when it's expected to be a list of daemon types. We could also consider just having a new function in HostCache `get_daemons_by_types` that handles getting daemons of multiple types at once, since all it's doing is repeatedly calling to the HostCache with a different type.

Reply:
@adk3798 I think it's a good idea. I created the new method and updated the code to use it. All the changes related to the review are packed in the last commit. I'll be squashing the changes later once all the comments are solved.